### ISSUE TITLE
spider dataset evaluation

### ISSUE THREAD

**Issue by chuangchuang-wang** [defog-ai/sqlcoder#12]
Has anyone tested the effect on the spider dataset? I have tested the result and the accuracy rate is only 30%, is it normal

**Comment by rishsriv** [comment]
Hi there, the Spider dataset uses SQLite while our model is trained to generate PostgreSQL. Additionally, our model is optimized for enterprise use-cases when columns descriptions are well defined. You can read more about the differences in the Spider dataset and our [SQL-Eval](https://github.com/defog-ai/sql-eval) here: https://defog.ai/blog/open-sourcing-sqleval/

**Comment by lihaoyang-ruc** [comment]
Hi @chuangchuang-wang ,

I personally think that one of the fundamental reasons for SQLCoder's poor performance on Spider is its lack of fine-tuning on Spider's training dataset (though this is speculative, and conclusions will have to wait until defog AI releases information about the data used for fine-tuning). However, it's undeniable that both ChatGPT and GPT-4 have demonstrated strong performance on Spider and Bird, even in zero-shot settings. This indicates that there's still a long way to go for the open-source community.

By the way, in order to compete with closed-source models, we introduced CodeS [(github page link)](https://github.com/RUCKBReasoning/codes), a language model specifically optimized for SQL generation. We conducted experiments with few-shot in-context learning and fine-tuning on Spider and Bird. While the few-shot performance of CodeS still lags behind closed-source models, we found that once fine-tuned on specific datasets, CodeS can even outperform prompt-based GPT-4. We have open-sourced all pre-training and fine-tuning checkpoints, and you are welcome to try them out at your convenience!
